permissions_validators/src/private_blockchain/mod.rs

```diff
@@ -38,3 +38,633 @@ impl<W: WorldTrait> IsGrantAllowed<W> for ProhibitGrant {
+                    .is_ok());
+                assert!(only_accounts_domain
+                    .check(&alice_id, &find_bronze, &wsv)
+                    .is_err());
+            }
+
+            {
+                let only_accounts_data: IsQueryAllowedBoxed<World> = query::OnlyAccountsData.into();
+
+                assert!(only_accounts_data
+                    .check(&alice_id, &find_gold, &wsv)
+                    .is_err());
+                assert!(only_accounts_data
+                    .check(&alice_id, &find_silver, &wsv)
+                    .is_err());
+                assert!(only_accounts_data
+                    .check(&carol_id, &find_bronze, &wsv)
+                    .is_err());
+                assert!(only_accounts_data
+                    .check(&alice_id, &find_bronze, &wsv)
+                    .is_err());
+            }
+        }
+
+        #[test]
+        fn find_assets_by_name() {
+            let TestEnv {
+                alice_id,
+                carol_id,
+                wsv,
+                ..
+            } = TestEnv::new();
+
+            let find_gold = QueryBox::FindAssetsByName(FindAssetsByName::new(
+                Name::from_str("gold").expect("Valid"),
+            ));
+            let find_silver = QueryBox::FindAssetsByName(FindAssetsByName::new(
+                Name::from_str("silver").expect("Valid"),
+            ));
+            let find_bronze = QueryBox::FindAssetsByName(FindAssetsByName::new(
+                Name::from_str("bronze").expect("Valid"),
+            ));
+
+            {
+                let only_accounts_domain: IsQueryAllowedBoxed<World> =
+                    query::OnlyAccountsDomain.into();
+
+                assert!(only_accounts_domain
+                    .check(&alice_id, &find_gold, &wsv)
+                    .is_err());
+                assert!(only_accounts_domain
+                    .check(&alice_id, &find_silver, &wsv)
+                    .is_err());
+                assert!(only_accounts_domain
+                    .check(&carol_id, &find_bronze, &wsv)
+                    .is_err());
+                assert!(only_accounts_domain
+                    .check(&alice_id, &find_bronze, &wsv)
+                    .is_err());
+            }
+
+            {
+                let only_accounts_data: IsQueryAllowedBoxed<World> = query::OnlyAccountsData.into();
+
+                assert!(only_accounts_data
+                    .check(&alice_id, &find_gold, &wsv)
+                    .is_err());
+                assert!(only_accounts_data
+                    .check(&alice_id, &find_silver, &wsv)
+                    .is_err());
+                assert!(only_accounts_data
+                    .check(&carol_id, &find_bronze, &wsv)
+                    .is_err());
+                assert!(only_accounts_data
+                    .check(&alice_id, &find_bronze, &wsv)
+                    .is_err());
+            }
+        }
+    }
+
+    mod revoke_and_grant {
```

Comment:
I'd suggest to add more negative tests scenarios in this module tests.
I.e. `add_register_domains_permission_allows_registering_domain_with_right_token()` checks that Alice can register new domainsm, but doesn't check if anyone without permission token can do it too.

Reply:
Added